codice/ddf
# [BLOCKED] [2.13.x] DDF-4457 The file system storage provider should encrypt the content stored on disk

**BLOCKED ON [4275](https://github.com/codice/ddf/issues/4275)**


#### What does this PR do?
Updates the `FileSystemStorageProvider` to encrypt the content it writes to disk.

#### Who is reviewing it? 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@garrettfreibott
@stustison

#### How should this be tested?
1. Build DDF and install the standard profile.
1. Ingest a resource using Intrigue.
1. Query for that resource using Intrigue.
1. Verify that the result looks normal
    1. Check for a thumbnail (if applicable)
    1. Verify all of the `Actions` in the `Inspector` tab work, especially the `resource` action.
1. Verify that the content is encrypted on disk
   1. Navigate to `<DDF_HOME>/data/content/store/{first 3 characters of id}/{next three characters of id}/{id}`, and verify that you cannot open the resource.
1. Verify that the newly generated Content Encryption Key has been added to the keystore.
    1. Navigate to the keystore directory under `<DDF_HOME>/etc/keystores`
    1. Run `keytool -list -keystore ./serverKeystore.jks`, give it the password `changeit`, and verify that there is a key with alias `content.encryption` added.

#### What are the relevant tickets?
[DDF-4457](https://codice.atlassian.net/browse/DDF-4457)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
